src/components/map/layers/edit/EditNetworkLayer.tsx

```diff
@@ -1,23 +1,76 @@
 import React, { Component } from 'react';
-import { Polyline } from 'react-leaflet';
+import * as L from 'leaflet';
+import { Polyline, withLeaflet } from 'react-leaflet';
+import { matchPath } from 'react-router';
 import { inject, observer } from 'mobx-react';
+import navigator from '~/routing/navigator';
+import SubSites from '~/routing/subSites';
 import { INode, ILink } from '~/models';
-import NodeLocationType from '~/types/NodeLocationType';
 import { EditNetworkStore } from '~/stores/editNetworkStore';
 import TransitTypeColorHelper from '~/util/transitTypeColorHelper';
 import NodeMarker from '../mapIcons/NodeMarker';
+import { LeafletContext } from '../../Map';
 
 interface IEditNetworkLayerProps {
     editNetworkStore?: EditNetworkStore;
+    leaflet: LeafletContext;
 }
 
 @inject('editNetworkStore')
 @observer
 class EditNetworkLayer extends Component<IEditNetworkLayerProps> {
+    editableLinks: L.Polyline[] = [];
+
+    private drawEditableLinks() {
+        const links = this.props.editNetworkStore!.links;
+        const map = this.props.leaflet.map;
+        const isLinkView = Boolean(matchPath(navigator.getPathName(), SubSites.link));
+        if (!isLinkView || !links || !map) return;
+
+        // Remove (possible) previously drawn links from map
+        this.editableLinks.forEach((editableLink) => {
+            map.removeLayer(editableLink);
+        });
+        this.editableLinks = [];
+
+        links.map((link: ILink, index) => this.drawEditableLinkToMap(link, index));
+
+        map.off('editable:vertex:dragend');
+        map.on('editable:vertex:dragend', () => {
+            const currentLinks = this.props.editNetworkStore!.links.map(link => ({ ...link }));
+            const latlngs = this.editableLinks[0].getLatLngs()[0] as L.LatLng[];
+            currentLinks[0].geometry = latlngs;
+            this.props.editNetworkStore!.setLinks(currentLinks);
+        });
+    }
+
+    private drawEditableLinkToMap(link: ILink, key: number) {
+        const map = this.props.leaflet.map;
+        if (map) {
+            const editableLink = L.polyline([link.geometry]).addTo(map);
+            editableLink.enableEdit();
+            const linkLatLngs = [editableLink.getLatLngs()];
+
+            // Hide first and last vertex from editableLink
+            linkLatLngs.forEach((_linkLatLng: any) => {
+                _linkLatLng.forEach((coords: any) => {
+                    const coordsToDisable = [coords[0], coords[coords.length - 1]];
+                    coordsToDisable.forEach((coordToDisable: any) => {
+                        const vertexMarker = coordToDisable.__vertex;
+                        vertexMarker.dragging.disable();
+                        vertexMarker.setOpacity(0);
+                    });
+                });
+            });
+            this.editableLinks.push(editableLink);
```

Comment:
This could be simplified slightly by using:

```
   private drawEditableLinkToMap(link: ILink) {
        const map = this.props.leaflet.map;
        if (map) {
            const editableLink = L.polyline([link.geometry]).addTo(map);
            editableLink.enableEdit();
            const links = editableLink.getLatLngs() as L.LatLng[][];
            const coords = links[0];
            const coordsToDisable = [coords[0], coords[coords.length - 1]];
            coordsToDisable.forEach((coordToDisable: any) => {
                const vertexMarker = coordToDisable.__vertex;
                vertexMarker.dragging.disable();
                vertexMarker.setOpacity(0);
            });

            this.editableLinks.push(editableLink);
        }
    }
```

editable.getLatLngs() will in our case always return L.LatLngs[][]. This is a list of line sections, but in our case the line will always only be in one section. So, we can extract our line with:

```
const links = editableLink.getLatLngs() as L.LatLng[][];
const coords = links[0];
```

Reply:
This was a nice simplification, thanks